### DISCUSSION TITLE
Can't set timezone

### DISCUSSION THREAD

**Discussion by Voroxpete** [haugene/docker-transmission-openvpn#2413] [Q&A]
My container keeps crashing on startup.  I'm still working on figuring out exactly why, but the first thing I noticed is that when I enable timestamps on my logs, everything comes through as UTC, not EST.

I've added the TZ=EST environment variable, and I've also tried passing through localtime from the host (which I've verified is set correctly) using /etc/localtime:/etc/localtime:ro volume mapping.  Nothing changes, I still get all my logs coming through in UTC.

Here's what I actually get in the logs if that helps:

2022-11-17T00:40:03.440088141Z Starting container with revision: b33d0fe4c938259a0d4eb844e55468f387456121,
2022-11-17T00:40:03.688329341Z WARNING: initial DNS resolution test failed,
2022-11-17T00:40:03.688372326Z Creating TUN device /dev/net/tun,
2022-11-17T00:40:03.688367256Z /etc/openvpn/start.sh: line 34:    11 Aborted                 (core dumped) nslookup ${HEALTH_CHECK_HOST:-"google.com"} > /dev/null 2>&1,
2022-11-17T00:40:03.696025546Z Using OpenVPN provider: NORDVPN,
2022-11-17T00:40:03.696172922Z Running with VPN_CONFIG_SOURCE auto,
2022-11-17T00:40:03.696192647Z Provider NORDVPN has a bundled setup script. Defaulting to internal config,
2022-11-17T00:40:03.696228842Z Executing setup script for NORDVPN,
2022-11-17T00:40:03.696937891Z INFO: OVPN: Checking curl installation,
2022-11-17T00:40:03.704934426Z thread.c:84: fatal error: pthread_create() failed: Operation not permitted

It keeps looping like this.  My suspicion is that the time problem is causing the connection to Nord to fail authentication, but since I can't get the clock to fix I don't really know how to proceed.  Trying to connect to the container doesn't work because it's constantly restarting.  Any help would be appreciated.

**Comment by pkishino** [comment]
please post complete docker config and system details.. that looks like something more serious

**Reply by Voroxpete** [reply]
version: '3.3'
services:
    transmission-openvpn:
        cap_add:
            - NET_ADMIN
        volumes:
            - deluge-public:/data
            - transmission:/config
        environment:
            - OPENVPN_PROVIDER=NORDVPN
            - OPENVPN_USERNAME=XXXXXXXXXXXXXXXXXXX
            - OPENVPN_PASSWORD=XXXXXXXXXXXXXXXXXXX
            - NORDVPN_COUNTRY=CA
            - NORDVPN_CATEGORY=legacy_p2p
            - NORDVPN_PROTOCOL=udp
            - LOCAL_NETWORK=XXXXXXXXXXXXXXX
            - WEBPROXY_ENABLED=true
            - WEBPROXY_PORT=8118
            - TRANSMISSION_WEB_UI=combustion
            - TZ=EST
        logging:
            driver: json-file
            options:
                max-size: 10m
        ports:
            - 9091:9091
            - 8118:8118
        image: haugene/transmission-openvpn
        restart: unless-stopped
        labels:
        - "com.centurylinklabs.watchtower.enable=true"
        
volumes:
    transmission:
    deluge-public:
        external: true


Not sure what system details you're looking for.  It's a linux box running Ubuntu 20.04.  It's running in a VM on a Proxmox host.

Some details:
Linux portainer 5.4.0-113-generic #127-Ubuntu SMP Wed May 18 14:30:56 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux)
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.3 LTS
Release:        20.04
Codename:       focal

Let me know if there's anything else I can provide.

**Comment by derekcentrico** [comment]
Please change your TZ=EST to the following:
      - TZ=America/New_York

**Reply by Voroxpete** [reply]
> Please change your TZ=EST to the following: - TZ=America/New_York

Tried that.  No difference.

**Comment by derekcentrico** [comment]
I see you're also missing this:
    volumes:
      - /etc/localtime:/etc/localtime:ro

**Reply by Voroxpete** [reply]
Sorry, still no change.  Exactly the same output, and the time is still UTC.

**Reply by Voroxpete** [reply]
I have also tried mapping /etc/timezone to the container as well, and that didn't make any difference either.

For the record, this is what timedatectl produces from the host:

host@portainer:~$ timedatectl
               Local time: Thu 2022-11-17 18:29:47 EST 
           Universal time: Thu 2022-11-17 23:29:47 UTC 
                 RTC time: Thu 2022-11-17 23:29:48     
                Time zone: America/Toronto (EST, -0500)
System clock synchronized: yes                         
              NTP service: active                      
          RTC in local TZ: no

Also, correction to an earlier statement, this host VM is on a VMware ESXi machine, not Proxmox.  I got mixed up between two servers.
